tests/unit/test_req.py

```diff
@@ -637,3 +665,96 @@ def test_mismatched_versions(caplog, tmpdir):
         'Requested simplewheel==2.0, '
         'but installing version 1.0'
     )
+
+
+@patch('pip._internal.req.req_install.os.path.isdir')
+@patch('pip._internal.req.req_install.os.path.isfile')
+def test_get_path_to_url(isdir_mock, isfile_mock):
+    isdir_mock.return_value = False
+    isfile_mock.return_value = False
+    name = 'foo @ git+http://foo.com@ref#egg=foo'
+    path = os.path.join('/path/to/' + name)
+    assert _get_path_to_url(path, name) is None
+
+
+@patch('pip._internal.req.req_install.os.path.isdir')
+@patch('pip._internal.req.req_install.os.path.isfile')
+def test_get_path_to_archive_pep440_url(isdir_mock, isfile_mock):
```

Comment:
Great to see you start writing these tests! A couple comments:

First, it's a helpful convention if when testing a function or method named `my_method`, the test function starts with the string `test_my_method`. That way it makes it easy to locate all the tests of a given function. So in this case, all of these should start with `test_get_path_to_url_...` (you don't need to include the leading underscore). Also, if you have more than one test function for a certain function, you can add a suffix describing the special case, like `test_get_path_to_url__archive_pep440_url()`. (I like to separate the function name portion from the suffix with a double underscore so someone can tell where the function name portion ends.)

Also, if you're testing multiple cases of a simple function, it helps to use `@pytest.mark.parametrize` to cut down on the amount of repetition. Take a look at `test_make_vcs_requirement_url` and the test functions following that for some examples. In this case, your inputs and outputs are strings (along with booleans to set your mocks), so it should be amenable to test parametrization.

Reply:
One more comment: I like to put the test functions in the same order as the original functions appear in the module. This also makes it easier to locate test functions when you're scrolling around. The test module has a parallel structure to the module it's testing.

Reply:
@cjerdonek is it ok to use a `noqa` on test names? Just in case they get too big.

Reply:
I don't expect they would ever get too big. You can put the arguments on the next line if it ever started to get too long.

Reply:
https://github.com/pypa/pip/pull/6203/commits/3e576738281350dee58242de25b1a735efd0ccc9
There, I had some issues parametrizing the tests that had URLs though.